src/basic/macro.h

```diff
@@ -488,4 +488,14 @@ static inline size_t size_add(size_t x, size_t y) {
         return y >= SIZE_MAX - x ? SIZE_MAX : x + y;
 }
 
+
+#define TRACE_POINT(name, dev, ...)                                             \
+        do {                                                                    \
+                const char *__s  = NULL;                                        \
+                DeviceAction __a = _DEVICE_ACTION_INVALID;                      \
+                (void) device_get_action(dev, &__a);                            \
```

Comment:
do make this double eval freee I think it would be better to copy the device ptr to a local variable too, and then use that for all calls, i.e. add `sd_device *_d = dev;` to the top of the block